Procyon.CompilerTools/src/main/java/com/strobel/decompiler/languages/java/ast/transforms/InsertNecessaryConversionsTransform.java

```diff
@@ -48,6 +51,7 @@
     }
 
     private final JavaResolver _resolver;
+    private final Deque<AstNode> replacements = new ArrayDeque<>();

```

Comment:
maybe a linked list, don't think array deque offers a benefit here, and it uses more memory for smaller stacks (default size 16)